core/src/main/java/com/epam/esm/dao/GiftDao.java

```diff
@@ -0,0 +1,20 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.dao.exception.DaoException;
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.entity.GiftCertificateEntity;
+
+import java.util.List;
+
+public interface GiftDao {
+    List<GiftCertificateEntity> findAllGifts();
+    GiftCertificateEntity findGiftById(Long giftId) throws DaoException;
+    List<GiftCertificateEntity> findAndSortGift(CustomSearchRequest customSearchRequest);
```

Comment:
better to use such naming of class as findAllGifts
just more usual for such cases